src/solvers/bicgstab_solver.hpp

```diff
@@ -105,37 +104,33 @@ class BiCGSTABSolver {
           solver->residual.val = 0.0;
           return TaskStatus::complete;
         },
-        this, i);
-    region.AddRegionalDependencies(reg_dep_id, i, initialize);
-    reg_dep_id++;
-    if (i == 0) {
-      tl.AddTask(dependence, [&]() {
-        if (Globals::my_rank == 0)
-          printf("# [0] v-cycle\n# [1] rms-residual\n# [2] rms-error\n");
-        return TaskStatus::complete;
-      });
-    }
+        this);
+    tl.AddTask(TaskQualifier::once_per_region, dependence, [&]() {
+      if (Globals::my_rank == 0)
+        printf("# [0] v-cycle\n# [1] rms-residual\n# [2] rms-error\n");
+      return TaskStatus::complete;
+    });
 
     // BEGIN ITERATIVE TASKS
+    auto [itl, solver_id] = tl.AddSublist(initialize, {1, params_.max_iters});
 
     // 1. u <- M p
-    auto precon1 = initialize;
+    auto precon1 = none;
```

Comment:
So now in an iterative list, the first task should always have `none` as a dependency, since the `AddSublist` arguments take in a dependency themself.  

Does this seem transparent to the downstream user? I wonder if we should provide a `itl.begin` `TaskID` that is `==` `none`.

Reply:
I don't think it has to have `none`, but it often will.  There's perhaps a little misdirection going on, but for various reasons this ended up being the cleanest solution IMHO.  I think it should work identically if instead of `none` you use the same dependencies as the sublist, but giving the sublist a dependency was necessary (the details of why are now a little foggy for me, but I could probably figure it out quickly if we really want to know).

I'm not sure I follow your suggestion about `itl.begin`.